dagger/dagger/main.go

```diff
@@ -0,0 +1,53 @@
+// A generated module for Dagger functions
+//
+// This module has been generated via dagger init and serves as a reference to
+// basic module structure as you get started with Dagger.
+//
+// Two functions have been pre-created. You can modify, delete, or add to them,
+// as needed. They demonstrate usage of arguments and return types using simple
+// echo and grep commands. The functions can be called from the dagger CLI or
+// from one of the SDKs.
+//
+// The first line in this comment block is a short description line and the
+// rest is a long description with more detail on the module's purpose or usage,
+// if appropriate. All modules should have a short description.
+
+package main
+
+import (
+	"context"
+	"dagger/dagger/internal/dagger"
+	"fmt"
+)
+
+type Dagger struct {
+	commandMap map[string][]string
+}
+
+func NewDagger() *Dagger {
+	return &Dagger{
+		commandMap: map[string][]string{
+			"evm":    {"bash", "-c", "(python3 /faucet.py &) ; /root/hardhat-boilerplate/start_node.sh"},
+			"solana": {"bash", "-c", "(python3 /faucet.py &) ; solana-test-validator --ticks-per-slot 32 --rpc-port ${RPC_PORT}"},
+		},
+	}
+}
+
+func (m *Dagger) NodeService(image string) *dagger.Service {
+	return dag.Container().
+		From(image).
+		WithWorkdir("/srv").
+		WithNewFile("index.html", "Hello, world!").
+		WithExec([]string{"bash", "-c", "(python3 /faucet.py &) ; /root/hardhat-boilerplate/start_node.sh"}).
```

Comment:
this is specific to EVM, likely you can just re-use the existing `CMD` from the image.